images/macos/provision/core/golang.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -e -o pipefail
+source ~/utils/utils.sh
+
+DEFAULT_GO_VERSION=$(get_toolset_value '.go.default')
+echo "Installing Go..."
+brew_smart_install "go@${DEFAULT_GO_VERSION}"
+
+# Create symlinks for Go 1.15 to preserve backward compatibility
+ln -sf $(brew --prefix go@${DEFAULT_GO_VERSION})/bin/* /usr/local/bin/
```

Comment:
the problem is that we need to create a symlink not only for 1.15 but for any non-latest go version. We have to either change a comment here or create a condition to check whether the installed version is the same as `brew install go` or not